File: SodaBuilder.ts
class SodaBuilder {
    private factory: SodaFactory;
    private bottle: Bottle;

    constructor(factory: SodaFactory) {
        this.factory = factory;
        this.bottle = factory.createBottle();
    }

    public pourSoda(): void {
        this.bottle.pourSoda(this.factory.createSoda());
    }

    public putLabel(): void {
        this.bottle.putLabel(this.factory.createLabel());
    }

    public seal(): void {
        this.bottle.seal(this.factory.createCloser());
    }

    public build(): Bottle {
        return this.bottle.copy();
    }
}

File: тз/SodaBuilder.java
package ru.agentd;

public class SodaBuilder {
    private SodaFactory factory;
    private Bottle bottle;

    public SodaBuilder(SodaFactory factory) {
        this.factory = factory;
        bottle = factory.createBottle();
    }

    public void pourSoda() {
        bottle.pourSoda(factory.createSoda());
    }

    public void putLabel() {
        bottle.putLabel(factory.createLabel());
    }

    public void seal() {
        bottle.seal(factory.createCloser());
    }

    public Bottle build() {
        return bottle.copy();
    }
}
